interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+            _printerr(err)
+            _printerr('\n| Line |')
+
+            for i in range(max(lineno-6, 0),
+                           min(lineno+3, len(text))):
+                if i == lineno-1:
+                    _printerr('> % 4i >' % (i+1), text[i].rstrip())
+                else:
+                    _printerr('| % 4i |' % (i+1), text[i].rstrip())
+        else:
+            # Error in ctml_writer or elsewhere
+            traceback.print_exc()
+
+        sys.exit(4)
+
+    # write the YAML file
+    emitter = yaml.YAML()
+    emitter.width = 70
+
+    for name, cls in globals().items():
+        if hasattr(cls, 'to_yaml'):
+            emitter.register_class(cls)
+
+    with open(outName, 'w') as dest:
+        # information regarding conversion
+        metadata = BlockMap([
+            ('generator', 'cti2yaml'),
+            ('cantera-version', '2.5.0a2'),
+            ('date', formatdate(localtime=True)),
+        ])
+        if filename is not None:
+            metadata['input-files'] = FlowList([base])
+        emitter.dump(metadata, dest)
+
+        outUnits = FlowMap([])
+        if _umass != 'kg':
+            outUnits['mass'] = _umass
+        if _ulen != 'm':
+            outUnits['length'] = _ulen
+        if _utime != 's':
+            outUnits['time'] = _utime
+        if _upres != 'Pa':
+            outUnits['pressure'] = _upres
+        if _uenergy != 'J':
+            outUnits['energy'] = _uenergy
+        if _umol != 'kmol':
+            outUnits['quantity'] = _umol
+        if _ue != 'J/kmol':
+            outUnits['activation-energy'] = _ue
+
+        if outUnits:
+            unitsMap = BlockMap([('units', outUnits)])
+            unitsMap.yaml_set_comment_before_after_key('units', before='\n')
+            emitter.dump(unitsMap, dest)
+
+        if _elements:
+            elementsMap = BlockMap([('elements', _elements)])
+            elementsMap.yaml_set_comment_before_after_key('elements', before='\n')
+            emitter.dump(elementsMap, dest)
+
+        if _phases:
+            phasesMap = BlockMap([('phases', _phases)])
+            phasesMap.yaml_set_comment_before_after_key('phases', before='\n')
+            emitter.dump(phasesMap, dest)
+
+        if _species:
+            speciesMap = BlockMap([('species', _species)])
+            speciesMap.yaml_set_comment_before_after_key('species', before='\n')
+            emitter.dump(speciesMap, dest)
+
+        for name, reactions in _reactions.items():
+            if reactions:
+                reactionsMap = BlockMap([(name, reactions)])
+                reactionsMap.yaml_set_comment_before_after_key(name, before='\n')
+                emitter.dump(reactionsMap, dest)
+
+
+def main():
+    if len(sys.argv) == 1  or sys.argv[1] in ('-h', '--help'):
+        print(__doc__)
+        sys.exit(0)
+    if len(sys.argv) not in (2,3):
+        raise ValueError('Incorrect number of command line arguments.')
```

Comment:
Should maybe print the proper function usage here instead of just telling the user they did it wrong.